File: library/src/main/java/io/github/anderscheow/validator/rules/regex/AlphanumericRule.java
package io.github.anderscheow.validator.rules.regex;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import io.github.anderscheow.validator.rules.common.RegexRule;

public class AlphanumericRule extends RegexRule {

    private static final String ALPHANUMERIC_REGEX = "(?=.*[a-zA-Z])(?=.*[\\d]).+";

    public AlphanumericRule() {
        super(ALPHANUMERIC_REGEX, "Value does not match alphanumeric regex");
    }

    public AlphanumericRule(@StringRes int errorRes) {
        super(ALPHANUMERIC_REGEX, errorRes);
    }

    public AlphanumericRule(@NonNull String errorMessage) {
        super(ALPHANUMERIC_REGEX, errorMessage);
    }
}

File: library/src/main/java/io/github/anderscheow/validator/rules/regex/AlphanumericRule.kt
package io.github.anderscheow.validator.rules.regex

import androidx.annotation.StringRes
import io.github.anderscheow.validator.rules.common.RegexRule

class AlphanumericRule : RegexRule {

    constructor(@StringRes errorRes: Int) : super(ALPHANUMERIC_REGEX, errorRes)

    constructor(errorMessage: String) : super(ALPHANUMERIC_REGEX, errorMessage)

    companion object {
        private const val ALPHANUMERIC_REGEX = "^[a-zA-Z0-9]*$"
    }
}

fun alphanumericOnly(@StringRes errorRes: Int): AlphanumericRule = AlphanumericRule(errorRes)

fun alphanumericOnly(errorMessage: String): AlphanumericRule = AlphanumericRule(errorMessage)